core/src/main/java/com/linecorp/armeria/internal/common/Http1ObjectEncoder.java

```diff
@@ -145,18 +145,15 @@ public void operationComplete(Future<Void> ignore) throws Exception {
                     final Throwable firstCause = first.cause();
                     final Throwable secondCause = second.cause();
 
-                    Throwable combinedCause = null;
-                    if (firstCause != null) {
-                        combinedCause = firstCause;
-                    }
-                    if (secondCause != null) {
-                        if (combinedCause == null) {
-                            combinedCause = secondCause;
-                        } else {
-                            combinedCause.addSuppressed(secondCause);
+                    final Throwable combinedCause;
+                    if (firstCause == null) {
+                        combinedCause = secondCause;
+                    } else {
+                        if (secondCause != null && secondCause != firstCause) {
+                            firstCause.addSuppressed(secondCause);
                         }
+                        combinedCause = firstCause;
```

Comment:
Super nit: What do you think of using `==` that might be easier to read?
```java
if (firstCause == null || firstCause == secondCause) {
    combinedCause = secondCause;
} else {
   if (secondCause != null) {
       firstCause.addSuppressed(secondCause);
   }
   combinedCause = firstCause;
}
```

Reply:
Not strong here. The review could be ignored.

Reply:
Then I will ignore :D feel free to make other style changes after this.